src/economy/shops/taxes.ts

```diff
@@ -0,0 +1,228 @@
+import { color, text } from 'craftjs-plugin/chat';
+import { UUID } from 'java.util';
+import { round } from 'lodash';
+import { ChatColor } from 'net.md_5.bungee.api';
+import { Bukkit, OfflinePlayer } from 'org.bukkit';
+import { Block, Chest, Sign } from 'org.bukkit.block';
+import { WallSign } from 'org.bukkit.block.data.type';
+import { Action, SignChangeEvent } from 'org.bukkit.event.block';
+import { PlayerInteractEvent } from 'org.bukkit.event.player';
+import * as yup from 'yup';
+import { errorMessage } from '../../chat/system';
+import { dataType } from '../../common/datas/holder';
+import { dataView } from '../../common/datas/view';
+import {
+  distanceBetween,
+  yupLocToLoc,
+  YUP_LOCATION,
+} from '../../common/helpers/locations';
+import { Currency, getShopCurrency, YUP_CURRENCY } from '../currency';
+import { giveMoney } from '../money';
+import { getBlockBehind } from './helpers';
+
+/**
+ * Calculate the amount of the tax and round it to 2 decimals points.
+ * @param taxRate Tax rate as a percentage from 0 to 100
+ * @param price Price including tax
+ * @returns Amount of the tax
+ */
+export function getTaxes(taxRate: number, price: number) {
+  const taxes = (taxRate / (100 + taxRate)) * price;
+  return round(taxes, 2);
+}
+
+const TAX_CHEST_DATA = {
+  taxCollector: yup.string().notRequired(),
+};
+
+const TAX_COLLECTOR_DATA = {
+  /**
+   * Location of the tax chest
+   */
+  chestLocation: YUP_LOCATION,
+
+  /**
+   * Array containing untransferred taxes for each currency
+   */
+  untransferred: yup
+    .array(
+      yup.object({
+        currency: YUP_CURRENCY.required(),
+        amount: yup.number().default(0),
+      }),
+    )
+    .default([]),
+};
+
+const TaxChestData = dataType('tax-chest-data', TAX_CHEST_DATA);
+const TaxCollectorData = dataType('tax-chest-data', TAX_COLLECTOR_DATA);
+
+export function sendTaxes(
```

Comment:
JSDoc would be nice. It seems that this only puts money to untransfered taxes and doesn't actually send them to a physical container anywhere.